lib/mayaUsd/ufe/UsdAttributes.cpp

```diff
@@ -454,8 +460,94 @@ bool UsdAttributes::doRemoveAttribute(const UsdSceneItem::Ptr& item, const std::
     }
     return false;
 }
+bool UsdAttributes::canRenameAttribute(
+    const UsdSceneItem::Ptr& sceneItem,
+    const std::string&       targetName,
+    const std::string&       newName)
+{
+    if (!sceneItem || !sceneItem->prim().IsActive()
+        || !UsdAttributes(sceneItem).hasAttribute(targetName)) {
+        return false;
+    }
+
+    // Do not rename if the newName is same as the targetName
+    if (targetName == newName) {
+        return false;
+    }
+
+    // Do not rename if an attribute already exists with the newName
+    if (UsdAttributes(sceneItem).hasAttribute(newName)) {
+        return false;
+    }
+
+    // We can rename attributes if they are:
+    // or boundary attributes,
+    // or custom attributes and not boundary
+
+    PXR_NS::TfToken nameAsToken(targetName);
+    auto            prim = sceneItem->prim();
+    auto            attribute = prim.GetAttribute(nameAsToken);
+
+    PXR_NS::UsdShadeNodeGraph      ngPrim(prim);
```

Comment:
Looks like some of this code could be replaced by a call to `canRemoveAttribute(targetName)` since the restrictions are exactly the same between removing and renaming.

Reply:
Yes exactly, I delete that function